### PR TITLE
Db setup cassandra

### PR DESCRIPTION
**Pull request by LoganAHart**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,7 @@ config/
 
 #DB files
 csv/
+project-notes/
+
+#Testing files
+artillery-logs/
```
Comment (tonaramos) [inline .gitignore:37]: new line at the end.

File: client/src/components/Overview.jsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ class Overview extends React.Component {
   getInitialData() {
     axios.get(`/overviews/restaurant/${this.props.match.params.restaurantId}/overview`)
       .then((response) => {
+        console.log(response);
```
Comment (tonaramos) [inline client/src/components/Overview.jsx:24]: There is a console.log left here.

File: db-cassandra/index.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  }
+  return postReqData;
+};
+
+//`COPY greyscale_overviews2.overview_by_id (rest_id, address, breakfast, breakfast_end, breakfast_start, city, cross_street, cuisine, description, dining_style, dinner, dinner_end, dinner_start, dress_code, executive_chef, lat, lgn, lunch, lunch_end, lunch_start, neighborhood, parking_details, payment_options, phone_number, price_range, public_transit, rest_name, tags, state, website, zip) FROM './csv/cass/overviewStreamID.csv' WITH DELIMITER='|'`
+
+const querySelectRestID = 'SELECT rest_id, address, breakfast, breakfast_end, breakfast_start, city, cross_street, ' +
+  'cuisine, description, dining_style, dinner, dinner_end, dinner_start, dress_code, executive_chef, lat, lgn, lunch, ' +
+  'lunch_end, lunch_start, neighborhood, parking_details, payment_options, phone_number, price_range, ' +
+  'public_transit, rest_name, tags, state, website, zip FROM overview_by_id WHERE rest_id = ?';
```
Comment (tonaramos) [inline db-cassandra/index.js:91]: Are these all of your columns? could you have used 'Select  * FROM overview_by_id WHERE rest_id = ?' ?

File: db-cassandra/index.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      return err;
+    } else if (err.message === 'bad request: 400') {
+      res.sendStatus(400);
+      return err;
+    }
+    return err;
+  }
+};
+
+
```
Comment (tonaramos) [inline db-cassandra/index.js:158]: Extra empty lines.

File: db-cassandra/index.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+const insertIntoDB = async (req, res, next) => {
+  try {
+    const rest_id = checkParams(req);
+    const postReqData = checkPostRequestBody(req);
+    const rows = await client.execute(queryInsertRow, postReqData, { prepare: true });
+    console.log(rows);
+    return rows;
+  } catch(err) {
+    //409 (conflict) if id already exists
```
Comment (tonaramos) [inline db-cassandra/index.js:146]: Will you be using this status code?

File: db-cassandra/index.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  }
+};
+
+
+
+module.exports = {
+  selectByID,
+  updateByID,
+  insertIntoDB,
+}
```
Comment (tonaramos) [inline db-cassandra/index.js:164]: New Line at end of page.

File: server/api/routes/restaurant.js [hunk 7]
```diff
@@ -1,39 +1,42 @@
 const express = require('express');
-const db = require('../../../database/index');
+const db = require('../../../db-cassandra/index.js');
 
 const router = express.Router();
 
-
 //Get Read: 200(ok), 404(not found)
```
Comment (tonaramos) [inline server/api/routes/restaurant.js:6]: Can probably delete this comment now.

File: server/api/routes/restaurant.js [hunk 8]
```diff
@@ -1,39 +1,42 @@
...
+  db.selectByID(req, res)
+    .then((result) => {
+      res.status(200).json(result.rows);
+    })
+    .catch((error) => {
+      console.log(error.message || error);
+    });
 });
 
 //POST Create: 201(created), 404 (not found), 409 (conflict)
```
Comment (tonaramos) [inline server/api/routes/restaurant.js:17]: delete comments, 409 I believe was not implemented.

File: server/api/routes/restaurant.js [hunk 9]
```diff
@@ -1,39 +1,42 @@
...
+  db.updateByID(req, res)
+    .then((result) => {
+      res.status(200).json({ response: result });
+    })
+    .catch((error) => {
+      console.log(error.message || error)
+    })
 });
 
 //Delete: 200(ok), 404(not found)
```
Comment (tonaramos) [inline server/api/routes/restaurant.js:42]: Extra comments, or future todo?

File: stress-test.json [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+                "url": "/overviews/restaurant/{{$randomNumber(9000000, 10000000)}}/overview"
+              }
+            }
+          ],
+          "count": 100
+        }
+      ]
+    }
+  ]
+}
```
Comment (tonaramos) [inline stress-test.json:50]: Missing new line at the end.

**Review by tonaramos — APPROVED** [review]
Looks good overall, I pointed out some minor changes and new lines at end of file.

File: db-cassandra/index.js (hunk 3 above)
Comment (LoganAHart) [reply db-cassandra/index.js:91]: The datastax website mentioned that the database doesn't have a mechanism for deleting prepared statements for select * if something drastic were to change to the database ( https://docs.datastax.com/en/developer/java-driver/3.0/manual/statements/prepared/#avoid-preparing-select-queries ) We probably won't be dropping any columns/changing things drastically though, so I guess it isn't all that relevant/was a lot of extra work. Thanks for the review Tona!

**Pull request merged**